### ISSUE TITLE
Android P Compatibility Warning... 

### ISSUE THREAD

**Issue by Ibuprophen** [ltguillaume/droidshows#65]
Hello Again My Friend! :-) 

I just wanted to provide you with a heads up with something that I've encountered a few of my developments and noticed the same for the DroidShows app as well. 

Simply put, when launching the DroidShows App in Android P, there has been a Popup message reflecting the App being for an earlier version of Android. This doesn't mean that it won't work for Android P. It's just a little bothersome to see (personally). 

I do know that it's tied in to the API/SDK of the app not targeting (or reflecting a minimum) pointing to the Android P but, I'm still trying to locate a definite solution for my developments. 

I figured that I should mention this to you before others ask about this sporadically over time. 

Thank you very much for your time and understanding with this! :-) 

~Ibuprophen

**Comment by ltguillaume** [comment]
Yeah I heard, but I also heard it's a one-time warning, right? I'll see if I can do anything about it. I really fucked up development, simply because I couldn't get saving to external sd cards working the way I wanted it, which also stalled a couple of other features I wanted to implement since two Christmases ago...

**Comment by Ibuprophen** [comment]
So far, regarding one of my developments (The SELinux Switch App), I was only able to prevent the message was to modify the source to Pie Only (API/SDK to 28/9) but, I'm still doing what I can to prevent this specific restriction on the app. 

It would be such a headache to be forced to branch off this development in such a way. 

I'll let you know, on my end, if I've made any progress so, hopefully, it will be helpful for you on your end. 

One more thing. The popup may be a one time message but, it's also affecting the way that the Pie provides the popup message regarding permissions that requires the individual to have to manually locate the appropriate permissions within the Privacy Guard. 

Privacy Guard provides permission options not available under the typical locations they were located in on the predecessors of the Android P. 


~Brian

**Comment by ltguillaume** [comment]
Shouldn't `android:targetSdkVersion="26"` in AndroidManifest.xml do the trick?

I have no clue (yet) what you're talking about concerning the privacy guard, though.

**Comment by Ibuprophen** [comment]
I'm still working on it but, regarding your app, that may just do the trick. 

It would just require testing on an Android P device. 

My development would probably require something a bit more. 

android:minSdkVersion="5"
android:targetSdkVersion="23"
android:maxSdkVersion="28"

I'm still working on this on my app... 


~Ibuprophen

**Comment by Ibuprophen** [comment]
@ltGuillaume, I've figured it out with my apps compatability for Pie and I receive no Popup warnings at all regarding the app being for an earlier version of Android. 

I had taken a quick look at the DroidShows app and it looks like the following should work for your app... 

android:minSdkVersion="9" android:targetSdkVersion="20"
android:maxSdkVersion="28" 

... That maintains its compatibility for the SDK9 while also providing its compatibility with SDK28 as well. 

I hope that this helps you out and I've got no problem with testing it out on my GS5 device with the Official LineageOS 16.0 klte build. 

Thank you very much My Friend! :-) 


~Ibuprophen

**Comment by ltguillaume** [comment]
Would you be so kind to test both of the attached APK's? I remember that I set the targetSDK to 11 in order to force the action bar, so I'm hoping it doesn't have to be changed...

**Edit**: should be fine without targetSDK = 11 (maybe because I added the drop-down menus in later versions, so that trick isn't necessary anymore). Anyway, hoping either of these versions work.

[droidshows_7.7.4-sdktests.zip](https://github.com/ltGuillaume/DroidShows/files/2999845/droidshows_7.7.4-sdktests.zip)

**Comment by Ibuprophen** [comment]
Not a problem at all My Friend! 

I'll try each of them one at a time and I'll do my best to be thorough. 

I'll go through all the features (I can think of/locate) that's available in the app to be sure it's working as designed. 

I'll get back to you as soon as I can, hopefully, with a good to go or a logcat of any unexpected issues. 

Thanks you for your time and support with this awesome app! :-) 


~Ibuprophen

**Comment by ltguillaume** [comment]
No no don't worry about any of the features, just test for this one thing: if newly installed, does it trigger the warning you're talking about? That's all I'm asking :)

**Comment by Ibuprophen** [comment]
@ltGuillaume, I had first tested the "Target 11" build (on 2 Android Devices with Pie) and it popped up the following Compatibility Warning... 

[DroidShows_Target-11](https://user-images.githubusercontent.com/18060271/54882615-379edc00-4e32-11e9-901d-f4aebf705fa8.png)

... I then uninstalled it and tested the "Target 20" build (on both devices) and there were no warnings that popped up at all. 

This is Great News since at least one of them is fully pie compatible. 

Just an FYI... the following image reflects what the DroidShows app looks like within the App Drawer of pie. 

[DroidShows_App-Drawer](https://user-images.githubusercontent.com/18060271/54882621-438a9e00-4e32-11e9-813a-2bdaa7c4f3c4.png)

The "Target 20" build works great on the Android Marshmallow, Nougat and Oreo as well (having other devices with them for testing purposes). 

Thanks a bunch and I'll be awaiting an Officially released version update! 

I won't mention this on XDA and leave it to you (the Developer) of course. 

If you require anything further from me, you know how to get in touch with me. 


~Ibuprophen

**Comment by ltguillaume** [comment]
Thanks for testing! That turned out to be easy! :)

**Comment by Ibuprophen** [comment]
@ltGuillaume, I had posted the following on XDA... 

https://forum.xda-developers.com/showthread.php?p=79203074 

... Regarding my opening the Beta Testing for The ClaRetoX2 Forum App (Info and screenshot examples provided there as well). 

I did notice that you had mentioned liking the The ClaRetoX Forum App and wanted to provide you with a heads up. 

If you're interested, I can PM (or email you) the link to the File Host directory to it. 

Thank you very much for your time and understanding with this. :-) 


~Ibuprophen
